### PR TITLE
add option to add files in container

### PR DESCRIPTION
**Pull request by mberndt123**
Hi,

I've added the possibility to add files that will be mounted into the scala-steward container. I needed this for two things:
 - for use with the `--repo-config` option which requires a file. I could of course put this into each individual repo, but I want to be able to configure it the same way across repos
 - for the `/root/.sbt/1.0/credentials.sbt` file where I read the Nexus credentials from the env variables. again, I don't want to modify lots of different repos to do this.

Let me know what you think.

### CODE REVIEW

File: charts/scala-steward/values.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ gitSecret: ""
 existingGitSecretName: ""
 
 image:
-  tag: 0.5.0-460-aad4fc41-SNAPSHOT
+  tag: latest-release
```
Comment (jsoref) [inline charts/scala-steward/values.yaml]: I think I want to split the version bump into its own commit (and probably not pick latest-release, but some pinned value). I'm assuming that `vcs` -> `forge` is a related change.

File: charts/scala-steward/templates/configmap.yaml [hunk 2]
```diff
@@ -11,3 +11,13 @@ data:
...
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "scala-steward.fullname" . }}-files
+data:
+  {{- range $k, $v := .Values.additionalFiles}}
+  {{ $k }}:
+    {{ $v.content | quote }}
+  {{- end}}
```
Comment (jsoref) [inline charts/scala-steward/templates/configmap.yaml]: I'd want to move this change...

File: charts/scala-steward/templates/cron-job.yaml [hunk 3]
```diff
@@ -43,6 +43,11 @@ spec:
                 name: opt
               - mountPath: /var/scala-steward/workspace
                 name: workspace
+              {{- range $k, $v := .Values.additionalFiles }}
+              - mountPath: {{ $v.path }}
+                name: files
+                subPath: {{ $k }}
+              {{- end }}
```
Comment (jsoref) [inline charts/scala-steward/templates/cron-job.yaml:50]: And this change...

File: charts/scala-steward/templates/cron-job.yaml [hunk 4]
```diff
@@ -63,3 +68,13 @@ spec:
...
+            {{- if .Values.additionalFiles }}
+            - name: files
+              configMap:
+                name: {{ include "scala-steward.fullname" . }}-files
+                items:
+                {{- range $k, $v := .Values.additionalFiles }}
+                  - key: {{ $k }}
+                    path: {{ $k }}
+                {{- end }}
+            {{- end }}
```
Comment (jsoref) [inline charts/scala-steward/templates/cron-job.yaml:80]: And this change...

File: charts/scala-steward/values.yaml [hunk 5]
```diff
@@ -53,3 +53,5 @@ persistence:
   accessModes:
     - ReadWriteOnce
   size: 5Gi
+
+additionalFiles: {}
```
Comment (jsoref) [inline charts/scala-steward/values.yaml]: ... And this change ... into the second commit.

File: charts/scala-steward/values.yaml [hunk 6]
```diff
@@ -54,4 +54,10 @@ persistence:
...
 
-additionalFiles: {}
\ No newline at end of file
+additionalFiles: {}
+# this can be used to make files available in the container, e. g. for authentication for an artifact repository
+# additionalFiles:
+#   nexusCredentials:
+#     path: /root/.sbt/1.0/credentials.sbt
+#     content: |
+#       credentials += Credentials(sys.env("NEXUS_REALM"), sys.env("NEXUS_HOST"), sys.env("NEXUS_USERNAME"), sys.env("NEXUS_PASSWORD"))
```
Comment (jsoref) [inline charts/scala-steward/values.yaml]: This change should be folded in with the commit that introduces `additionalFiles` (but not the commit that bumps the version).

**Review by jsoref — COMMENTED** [review]
This looks interesting. As it happens, we planned to look into upgrading scala steward this month...

**Comment by mberndt123** [comment]
Thanks for the review, I've rebased the changes and also adjusted the version from `latest-release` to `0.30.2`.

**Comment by mberndt123** [comment]
I've also adjusted `version` and `appVersion` in `Chart.yaml`

**Comment by mberndt123** [comment]
@jsoref can you take another look?

**Comment by jsoref** [comment]
Thanks! The changes _look_ right. I want to test this a bit before I merge. I'm busy today and am hoping to get a coworker to look soon (maybe tomorrow if I'm lucky). If I don't respond by Monday, please tag me on Tuesday.

**Comment by mberndt123** [comment]
OK cool

**Comment by mberndt123** [comment]
Ping @jsoref!

I've pushed another change to correctly handle the case where `additionalFiles` is empty.

**Comment by jsoref** [comment]
Fwiw, I've asked someone to run a test of this, so hopefully they'll check tomorrow and have feedback for me by Monday.

**Comment by mberndt123** [comment]
I'm running this right now and it's working fine for me (I've added a credentials configuration for sbt and a repo configuration for scala-steward), so I'm not sure what more testing it is you need to do. But go ahead, do your thing.

**Comment by jsoref** [comment]
Sorry... sometimes I'm overly cautious. I'm used to things going wrong. Many things I've tried this week have gone wrong.

What I can say is that we tested the basic chart from this PR and it worked. We haven't had time to test the volume mounts although I absolutely expect them to work.

I'm going to check with a coworker and I think I'll merge by Monday.

**Comment by mberndt123** [comment]
Thanks for merging @jsoref !

**Comment by jsoref** [comment]
Thanks for contributing and sorry about the long time to merge.
